app/scenes/Document/index.tsx

```diff
@@ -32,6 +35,16 @@ export default function DocumentScene(props: Props) {
 
   React.useEffect(() => () => ui.clearActiveDocument(), [ui]);
 
+  React.useEffect(() => {
+    // When opening a document directly on app load, sidebarContext will not be set.
```

Comment:
Oof, this is going to result in another render on every fresh document load :/

Reply:
When we navigate between documents (both fresh and cached), this won't re-render since the location state will have sidebar context..
I think we'll have an additional render only when we navigate from a non-collection and non-document space 🤔

Reply:
Feels okay in use... maybe we'll get away with it